File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int pos = 0;
        int neg = 0;
        int ans = INT_MIN;
        for(int i=0; i<nums.size(); i++){
            pos += nums[i];
            neg += nums[i];

            ans = max(ans,max(pos,neg));
            if(pos<0){
                pos=0;
            } 
        }
        return ans;
    }
};

File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum = 0;
        int negsum = Integer.MAX_VALUE;
        int answer = Integer.MIN_VALUE;
        for(int i=0; i<nums.length; i++){
            sum = sum+nums[i];
            negsum = nums[i];
            answer = Math.max(sum,answer);
            if(sum<0){
                sum = 0;
            }
            
        }
        return answer;
    }
}